readthedocsext/theme/templates/account/login.html

```diff
@@ -16,23 +16,42 @@
 
       <div class="ui medium header">
         {% block authentication_header %}
-          {% trans "Log in to continue" %}
+          {% if is_from_cas_login %}
+            {% trans "Log in to view this documentation" %}
+          {% else %}
+            {% trans "Log in to continue" %}
+          {% endif %}
         {% endblock authentication_header %}
       </div>
 
-      <div class="ui small stackable secondary two item menu" data-bind="semanticui: {tabs: {history: true, autoTabActivation: false}}">
+      <div
+          {# The password URL is listed as an additional tab if present #}
+          class="ui small stackable secondary item menu {% if project_password_url %}three{% else %}two{% endif %}"
+          data-bind="semanticui: {tabs: {history: true, autoTabActivation: false}}">
 
         {# This is purposely minimal, as the verbiage for "Log in with connected account" is a mountful #}
         <a class="item" data-tab="vcs">
           <i class="fad fa-cloud icon"></i>
           {% block authentication_vcs_text %}{% trans "Connected account" %}{% endblock %}
         </a>
 
-        <a class="item" data-tab="email">
+        {# If allowed providers is given, disable the username/password form #}
+        <a class="item {% if allowed_providers %}disabled{% endif %}"
+           data-tab="email"
+           {% if allowed_providers %}
+           title="Log in using email is not enabled for this organization"
+           aria-label="Log in using email is not enabled for this organization"
```

Comment:
:+1: for catching the aria label here. We'll probably tweak this pattern, but here are a few general notes for these patterns:

- I've been using `trans .. asvar ...` to localize text once. (Later we should probably just add a template filter or something).
- FomanticUI has a tooltip module that uses the attribute `data-content`. It's probably still not going to show for disabled items though.


```suggestion
           {% trans "Log in using email is disabled for this organization" asvar text_email_disabled %}
           data-content="{{ text_email_disabled }}"
           aria-label="{{ text_email_disabled }}"
```